Documents/Automation/EncryptRootVolume/Documents/npark-encryptrootvolume.json

```diff
@@ -223,72 +245,115 @@
           "InstanceId": "{{instanceId}}",
           "VolumeId": "{{createVolume.NewRootVolumeID}}"
         }
-      },
-      "name": "attachNewEBSVolume",
-      "action": "aws:executeAutomation",
-      "timeoutSeconds": 180,
-      "onFailure": "Abort"
+      }
     },
+
     {
-       "name": "applyDeleteOnTerminationValue",
-       "action": "aws:executeAwsApi",
-       "onFailure": "Abort",
-       "timeoutSeconds": 60,
-       "maxAttempts": 10,
-       "inputs": {
-         "Service": "ec2",
-         "Api": "ModifyInstanceAttribute",
-         "InstanceId": "{{instanceId}}",
-         "BlockDeviceMappings": [
-           {
-             "DeviceName": "{{describeInstance.rootDeviceName}}",
-             "Ebs": {
-               "DeleteOnTermination": "{{describeInstanceRootVolume.RootDeviceDeleteOnTermination}}"
-             }
-           }
-         ]
-       },
-       "isCritical": true
+      "name": "applyDeleteOnTerminationValue",
+      "action": "aws:executeAwsApi",
+      "onFailure": "step:detachNewVolume",
+      "timeoutSeconds": 60,
+      "maxAttempts": 10,
+      "isCritical": true,
+      "nextStep": "deleteRootVolumeSnapshot",
+      "inputs": {
+        "Service": "ec2",
+        "Api": "ModifyInstanceAttribute",
+        "InstanceId": "{{instanceId}}",
+        "BlockDeviceMappings": [
+          {
+            "DeviceName": "{{describeInstance.rootDeviceName}}",
+            "Ebs": {
+              "DeleteOnTermination": "{{describeInstanceRootVolume.RootDeviceDeleteOnTermination}}"
+            }
+          }
+        ]
+      }
     },
 
     {
       "name": "restoreInstanceInitialState",
```

Comment:
Question: we are not deleting the original root volume. We should probably mention that in the document description. If the root volume stays after the encryption process, you may also want to add it to the output of this document.

The overall output could containt:
- Original root volume ID
- New encrypted root volume ID

What do you think?

Reply:
added outputs section in the Document